File: test/integration/activate_test.py
import os
import tempfile

import auth_test
import helpers

dir_path = os.path.abspath(os.path.dirname(os.path.realpath(__file__)))

class TestActivate(helpers.IntegrationTest):

    def test_activate_python2(self):
        path = os.path.join(dir_path, "testdata")
        self.set_cwd(path)

        auth = auth_test.TestAuth()
        auth.set_config(self.config_dir)
        auth.set_cwd(path)
        auth.login_as_persistent_user()

        self.env["ACTIVESTATE_CLI_DISABLE_RUNTIME"] = "false"

        self.spawn("activate ActiveState-CLI/Python2")
        self.expect("Where would you like to checkout")
        self.send(tempfile.TemporaryDirectory().name)
        if os.name == 'nt':
            # No py2 project on the platform yet
            self.expect("Your project does not have a configuration that is compatible with your platform")
        else:
            self.expect("Downloading", timeout=240)
            self.expect("Installing", timeout=240)
            self.wait_ready(timeout=240)
            self.send("python2 -c \"import sys; print(sys.copyright)\"")
            self.expect("ActiveState Software Inc.")
            self.send("python2 -c \"import numpy; print(numpy.__doc__)\"")
            self.expect("import numpy as np")
        self.send_quit()
        if os.name == 'nt':
            # Windows returns nonzero on successful murdering of a process
            self.wait(code=1)
        else:
            self.wait()

    def test_activate_python3(self):
        path = os.path.join(dir_path, "testdata")
        self.set_cwd(path)

        auth = auth_test.TestAuth()
        auth.set_config(self.config_dir)
        auth.set_cwd(path)
        auth.login_as_persistent_user()

        self.env["ACTIVESTATE_CLI_DISABLE_RUNTIME"] = "false"

        self.spawn("activate ActiveState-CLI/Python3")
        self.expect("Where would you like to checkout")
        self.send(tempfile.TemporaryDirectory().name)
        self.expect("Downloading", timeout=240)
        self.expect("Installing", timeout=240)
        self.wait_ready(timeout=600)
        self.send("which python3")
        self.send("python3 -c \"import sys; print(sys.copyright)\"")
        self.expect("ActiveState Software Inc.")
        self.send("python3 -c \"import numpy; print(numpy.__doc__)\"")
        self.expect("import numpy as np")
        self.send_quit()
        if os.name == 'nt':
            # Windows returns nonzero on successful murdering of a process
            self.wait(code=1)
        else:
            self.wait()

if __name__ == '__main__':
    helpers.Run()

File: test/integration/activate_test.go
package integration_test

import (
	"io/ioutil"
	"os"
	"path/filepath"
	"testing"
	"time"

	"github.com/stretchr/testify/suite"

	"github.com/ActiveState/cli/internal/environment"
	"github.com/ActiveState/cli/test/integration/expect"
)

type ActivateTestSuite struct {
	Suite
}

func (suite *ActivateTestSuite) TestActivatePython3() {
	root := environment.GetRootPathUnsafe()
	os.Chdir(filepath.Join(root, "test/integration/testdata"))

	suite.LoginAsPersistentUser()
	suite.AppendEnv([]string{"ACTIVESTATE_CLI_DISABLE_RUNTIME=false"})

	tempDir, err := ioutil.TempDir("", "")
	suite.Require().NoError(err)
	os.Remove(tempDir)
	suite.Require().NoError(err)

	suite.Spawn("activate", "ActiveState-CLI/Python3")
	suite.Expect("Where would you like to checkout")
	suite.Send(tempDir)
	suite.Expect("Downloading")
	suite.Expect("Installing", 120*time.Second)
	suite.WaitForInput(120 * time.Second)
	suite.Send("echo \"PATH: $PATH\"")
	suite.Send("echo \"python3 bin: $(which python3)\"")
	suite.Send("echo \"tty: $(tty)\"")
	suite.Send("python3 -c \"import sys; print(sys.copyright)\"")
	suite.Expect("ActiveState Software Inc.")
	suite.Send("python3 -c \"import numpy; print(numpy.__doc__)\"")
	suite.Expect("import numpy as np")
	suite.SendQuit()
	suite.Wait()
}

func TestActivateTestSuite(t *testing.T) {
	_ = suite.Run // vscode won't show test helpers unless I use this .. -.-

	//suite.Run(t, new(ActivateTestSuite))
	expect.RunParallel(t, new(ActivateTestSuite))
}
